apigee_edge_ui.module

```diff
@@ -0,0 +1,170 @@
+  // Format the app list header based on template_preprocess_table().
+  if (!empty($variables['header'])) {
+    $ts = tablesort_init($variables['header']);
+
+    foreach ($variables['header'] as $col_key => $cell) {
+      if (!is_array($cell)) {
+        $cell_content = $cell;
+        $cell_attributes = new Attribute();
+      }
+      else {
+        $cell_content = '';
+        if (isset($cell['data'])) {
+          $cell_content = $cell['data'];
+          unset($cell['data']);
+        }
+
+        tablesort_header($cell_content, $cell, $variables['header'], $ts);
+
+        // tablesort_header() removes the 'sort' and 'field' keys.
+        $cell_attributes = new Attribute($cell);
+      }
+      $variables['header'][$col_key] = [];
+      $variables['header'][$col_key]['attributes'] = $cell_attributes;
+      $variables['header'][$col_key]['content'] = $cell_content;
+    }
+
+    $variables['header_attributes'] = new Attribute(['class' => 'developer-app-list__header']);
+  }
+
+  // Format the app list.
+  if (!empty($variables['apps'])) {
+    $variables['apps_attributes'] = new Attribute(['class' => 'developer-app-list__apps']);
+    $variables['show_text'] = t('Show details');
+    $variables['hide_text'] = t('Hide details');
+  }
+  else {
+    $variables['empty_attributes'] = new Attribute(['class' => 'developer-app-list__apps--empty']);
+  }
+}
+
+/**
+ * Implements hook_form_alter().
+ */
+function apigee_edge_ui_form_alter(array &$form, FormStateInterface $form_state, string $form_id): void {
+  $app_edit_form_ids = [
+    'developer_app_edit_for_developer_form',
+    'developer_app_edit_form',
+  ];
+
+  $app_add_form_ids = [
+    'developer_app_add_for_developer_form',
+    'developer_app_add_form',
+  ];
+
+  if (in_array($form_id, $app_add_form_ids)) {
+    if (isset($form['api_products'])) {
+      $form['api_products']['#attributes']['class'][] = 'developer-app-form__products';
+    }
+    $form['#attributes']['class'][] = 'developer-app-form--add';
+  }
+  elseif (in_array($form_id, $app_edit_form_ids)) {
+    $app_label =  \Drupal::service('entity_type.manager')->getDefinition('developer_app')->getSingularLabel();
+    $form['actions']['submit']['#value'] = t('Save @app', ['@app' => $app_label]);
+    $form['actions']['delete']['#title'] = t('Delete @app', ['@app' => $app_label]);
+    $form['actions']['delete']['#weight'] = 1;
+    $form['#attributes']['class'][] = 'developer-app-form--edit';
+
+    $detail_fields = [
+      'callbackUrl',
+      'description',
+      'displayName',
+    ];
+
+    // Add details wrapper to non-credential info fields.
+    $form['details'] = [
+      '#type' => 'html_tag',
+      '#tag' => 'div',
+      '#attributes' => [
+        'class' => [
+          'developer-app-form__details',
+        ],
+      ],
+    ];
+
+    foreach ($detail_fields as $field) {
+      $form['details'][$field] = $form[$field];
+      unset($form[$field]);
```

Comment:
Do not remove the old form element, because other alter hooks may rightfully expect that it is there. Isn't there the other way to _visually_ move form elements to $form['details']? For example in template processes?

Reply:
Preprocess was my first idea too, but it doesn't work because the the whole form is in a single markup string in the preprocess function so I can't modify the elements there. Do you have other suggestion? I can add `'#prefix` and `'#suffix'` properties to the first and last detail element but it didn't seem a nice solution.

Reply:
Hm, I do not have a better idea yet but instead of removing the element from the form array you may use the `hide()` function to hide the original form element. Or use the `#access = FALSE` trick.
https://api.drupal.org/api/drupal/core%21includes%21common.inc/function/hide/8.6.x